authors/apps/followers/views.py

```diff
@@ -0,0 +1,136 @@
+from .models import Follow
+from authors.apps.core.permissions import IsOwnerOrReadOnly
+from authors.apps.authentication.models import User
+from rest_framework.generics import (
+    ListCreateAPIView, DestroyAPIView)
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly, IsAuthenticated)
+from .serializers import (FollowSerializer,
+                          FollowerSerializer,
+                          FollowerListSerializer)
+from rest_framework.exceptions import ValidationError
+from .renderers import FollowerJsonRenderer, FollowerListJsonRenderer
+from rest_framework import status
+from rest_framework.response import Response
+
+
+def user_not_found():
+    raise ValidationError(
+        {'error': 'User with that username not found'})
+
+
+class ListCreateFollow(ListCreateAPIView):
+    queryset = Follow.objects.all()
+    serializer_class = FollowSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (FollowerJsonRenderer,)
+
+    def post(self, request, username):
+        """
+        A method for following a user
+        """
+        user_exists = User.objects.filter(username=username).exists()
+        if not user_exists:
+            return Response(
+                {'error': 'user with that name was not found'},
+                status.HTTP_404_NOT_FOUND)
+        # we check if the user is already followed
+        followed_user = User.objects.get(username=username)
+        already_followed = Follow.is_user_already_followed(
+            followed_user_id=followed_user.id,
+            user_id=self.request.user.id
+        )
+        if already_followed:
+            return Response({'error': 'user already followed'},
+                            status.HTTP_400_BAD_REQUEST)
+        if followed_user.id == self.request.user.id:
+            return Response({'error': "you cannot follow yourself."},
+                            status.HTTP_400_BAD_REQUEST)
+        data = {
+            "followed_user": followed_user.id,
+            "user": self.request.user.id}
+        serializer = FollowSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response({'message': 'user followed successfully'},
+                        status.HTTP_201_CREATED)
+
+    # def get(self, request):
+    #     queryset = Follower.objects.filter(user=self.request.user)
+    #     serializer = self.serializer_class(queryset, many=True)
+    #     return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class FollowersView(ListCreateAPIView):
```

Comment:
Since we are not implementing the `post` method here,and having a serializer class that inherits `rest_framework.serializers.Baseserializer`
 `class FollowerSerializer(serializers.BaseSerializer): ` 

```suggestion
class FollowersView(ListApiView):
```
This will, on attempt to send a post request to `../followers/` help us avoid
![ser_err](https://user-images.githubusercontent.com/44059971/55674853-b11edb80-58c2-11e9-97a2-78009703e982.jpg)